swift/Arrow/Tests/ArrowTests/ArrayTests.swift

```diff
@@ -212,6 +212,61 @@ final class ArrayTests: XCTestCase { // swiftlint:disable:this type_body_length
         XCTAssertEqual(microArray[2], 987654321)
     }
 
+    func testStructArray() throws {
+        class StructTest {
+            var field0: Bool = false
+            var field1: Int8 = 0
+            var field2: Int16 = 0
+            var field3: Int32 = 0
+            var field4: Int64 = 0
+            var field5: UInt8 = 0
+            var field6: UInt16 = 0
+            var field7: UInt32 = 0
+            var field8: UInt64 = 0
+            var field9: Double = 0
+            var field10: Float = 0
+            var field11: String = ""
+            var field12 = Data()
+            var field13: Date = Date.now
+        }
+
+        let testData = StructTest()
+        let dateNow = Date.now
+        let structBuilder = try ArrowArrayBuilders.loadStructArrayBuilderForType(testData)
+        structBuilder.append([true, Int8(1), Int16(2), Int32(3), Int64(4),
+                              UInt8(5), UInt16(6), UInt32(7), UInt64(8), Double(9.9),
+                              Float(10.10), "11", Data("12".utf8), dateNow])
+        structBuilder.append(nil)
+        structBuilder.append([true, Int8(13), Int16(14), Int32(15), Int64(16),
+                              UInt8(17), UInt16(18), UInt32(19), UInt64(20), Double(21.21),
+                              Float(22.22), "23", Data("24".utf8), dateNow])
+        XCTAssertEqual(structBuilder.length, 3)
+        let structArray = try structBuilder.finish()
+        XCTAssertEqual(structArray.length, 3)
+        XCTAssertNil(structArray[1])
+        XCTAssertEqual(structArray.arrowFields![0].length, 3)
+        XCTAssertNil(structArray.arrowFields![0].array.asAny(1))
+        XCTAssertEqual(structArray[0]![0] as? Bool, true)
+        XCTAssertEqual(structArray[0]![1] as? Int8, 1)
+        XCTAssertEqual(structArray[0]![2] as? Int16, 2)
+        XCTAssertEqual(structArray[0]![3] as? Int32, 3)
+        XCTAssertEqual(structArray[0]![4] as? Int64, 4)
+        XCTAssertEqual(structArray[0]![5] as? UInt8, 5)
+        XCTAssertEqual(structArray[0]![6] as? UInt16, 6)
+        XCTAssertEqual(structArray[0]![7] as? UInt32, 7)
+        XCTAssertEqual(structArray[0]![8] as? UInt64, 8)
+        XCTAssertEqual(structArray[0]![9] as? Double, 9.9)
+        XCTAssertEqual(structArray[0]![10] as? Float, 10.10)
+        XCTAssertEqual(structArray[2]![11] as? String, "23")
+        XCTAssertEqual(String(decoding: (structArray[0]![12] as? Data)!, as: UTF8.self), "12")
```

Comment:
Can we use field name instead of index such as `structArray[0]!["field0"]` for readability?